### PR TITLE
feat: vite

### PR DESCRIPTION
**Pull request by BerkieBb**
## Description

Fixes the app

## Checklist

- [x] I have personally loaded this code into an updated Qbox project and checked all of its functionality.
- [x] My pull request fits the contribution guidelines & code conventions.

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,9 +1,9 @@
 import React, { useEffect, useState } from 'react';
-import { NuiProvider } from 'react-fivem-hooks';
-import { useHistory } from 'react-router-dom';
+import { NuiProvider } from 'fivem-nui-react-lib';
```
Comment (itschip) [inline src/App.tsx]: You don't actually need this. Just use the `useNuiHook` from npwd (copy-paste) into the codebase. No need for the lib here.

File: src/App.tsx [hunk 2]
```diff
@@ -1,9 +1,9 @@
...
+import { useNavigate } from 'react-router-dom';
 import styled from 'styled-components';
```
Comment (itschip) [inline src/App.tsx]: Replace styled-components with `@emotion/styled`

File: src/App.tsx [hunk 3]
```diff
@@ -85,9 +88,11 @@ const App = (props: AppProps) => {
 };
 
 const WithProviders: React.FC<AppProps> = (props) => (
-  <NuiProvider>
-    <App {...props} />
-  </NuiProvider>
+  <RecoilRoot override key="npwd_qbx_garages">
+    <NuiProvider resource="npwd_qbx_garages">
```
Comment (itschip) [inline src/App.tsx]: As ur removing the nui lib, remove `NuiProvider`

File: vite.config.ts [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+      filename: "remoteEntry.js",
+      exposes: {
+        "./config": "./npwd.config.ts",
+      },
+      shared: [
+        "react",
+        "react-dom",
+        "@emotion/react",
+        "react-router-dom",
+        "fivem-nui-react-lib",
```
Comment (itschip) [inline vite.config.ts]: Remove `fivem-nui-react-lib`

**Review by itschip — CHANGES_REQUESTED** [review]

File: src/App.tsx [hunk 5]
```diff
@@ -52,20 +52,19 @@ const App = (props: AppProps) => {
...
-      const mappedVehicles = vehicles?.reduce((vehs: any, vehicle: any) => {
-        if (!vehs[vehicle.type]) vehs[vehicle.type] = [];
-        vehs[vehicle.type].push(vehicle);
-        return vehs;
-      }, {});
+    const mappedVehicles = new Map<string, GarageItem[]>()
+    vehicles.forEach((vehicle: GarageItem) => {
+      if (!mappedVehicles.get(vehicle.type)) mappedVehicles.set(vehicle.type, []);
+      mappedVehicles.get(vehicle.type)?.push(vehicle);
+    }, {});
```
Comment (itschip) [inline src/App.tsx]: You can just do this directly in the fetchNui promise chain :)

**Review by itschip — COMMENTED** [review]
LTGM

**Review by mafewtm — APPROVED** [review]

**Pull request merged**